framework/src/wifi_manager/wifi_manager.c

```diff
@@ -274,17 +300,18 @@ static wifi_manager_result_e wifi_fetch_scan_results(slsi_scan_info_t **wifi_sca
 static wifi_manager_result_e wifi_scan_result_callback(slsi_reason_t *reason)
 {
 	wifi_manager_scan_info_s *wifi_manager_scan_info = 0;
+	wifi_manager_cb_s *wifi_cb = g_manager_callback;
 	if (reason->reason_code == 0) {
 		slsi_scan_info_t *wifi_scan_result;
 		WiFiGetScanResults(&wifi_scan_result);
 		if (wifi_fetch_scan_results(&wifi_scan_result, &wifi_manager_scan_info) == WIFI_MANAGER_SUCCESS) {
-			g_manager_info.wmcb->scan_ap_done(&wifi_manager_scan_info, WIFI_SCAN_SUCCESS);
+			wifi_cb->scan_ap_done(&wifi_manager_scan_info, WIFI_SCAN_SUCCESS);
```

Comment:
i thingl it's better to call after NULL check

Reply:
null checkpoint is in wifi_manager_scan_ap().